core/src/test/java/nu/yona/server/subscriptions/service/BuddyDeviceChangeMessageDtoTest.java

```diff
@@ -193,7 +189,7 @@ public void managerHandleAction_deviceAdded_deviceExistsForBuddy()
 		manager.handleAction(UserDto.createInstance(bob), messageEntity, "process", null);
 
 		// Assert success
-		assertTrue(messageEntity.isProcessed());
+		assertThat("Message is not processed", messageEntity.isProcessed());
```

Comment:
This is not correct, should be `assertTrue(messageEntity.isProcessed(), () -> "Message is not processed")` or `assertThat(messageEntity.isProcessed(), is(true), () -> "Message is not processed")`, same for other places in this file

Reply:
Why is this wrong? It seems in line with [the documentation](http://hamcrest.org/JavaHamcrest/javadoc/2.1/).

Reply:
I don't see any example like in the code above in the [MatcherAssert doc](http://hamcrest.org/JavaHamcrest/javadoc/2.1/org/hamcrest/MatcherAssert.html). In the [Matchers doc](http://hamcrest.org/JavaHamcrest/javadoc/2.1/org/hamcrest/Matchers.html) and the [JUnit 5 user guide](https://junit.org/junit5/docs/current/user-guide/) there are many examples but not like the code above.

Reply:
I don't see examples indeed, but it's still valid code. See also [Stackoverflow here](https://stackoverflow.com/questions/23398736/custom-error-message-for-assertthat-in-junit).

Reply:
Let's stick to the manual :). The stackoverflow example has 3 arguments, not 2.

Reply:
The Stackoverflow example uses the second of [these methods](http://hamcrest.org/JavaHamcrest/javadoc/2.1/org/hamcrest/MatcherAssert.html#MatcherAssert--) while I'm using the first one. Why is using the first one wrong?

Reply:
I still don't really get your concern here. The ``assertThat`` API that I'm using here is a valid API, not deprecated and it works properly. To prove that, I've added an exclamation mark in front of ``messageEntity.isProcessed()``, and that results in a proper test failure:
```
java.lang.AssertionError: Message is not processed
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:26)
	at nu.yona.server.subscriptions.service.BuddyDeviceChangeMessageDtoTest.managerHandleAction_deviceAdded_deviceExistsForBuddy(BuddyDeviceChangeMessageDtoTest.java:192)
```

The ``assertThat`` operation is from Hamcrest, and we use it > 300 times. The ``assertTrue`` operation is from JUnit and we are not using it anymore after this refactoring. You are right that the one you refer to is an alternative, and there is even a simpler one. We could write the above statement this way:
```
assertTrue(messageEntity.isProcessed(), "Message is not processed");
```
If I make that fail like above, you get this:
```
org.opentest4j.AssertionFailedError: Message is not processed ==> expected: <true> but was: <false>
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:38)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:40)
	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:168)
	at nu.yona.server.subscriptions.service.BuddyDeviceChangeMessageDtoTest.managerHandleAction_deviceAdded_deviceExistsForBuddy(BuddyDeviceChangeMessageDtoTest.java:193)
```
This would be our first use of ``assertTrue`` and it would not be any better than the statement I wrote.

Reply:
OK, you're right it works, I just have never seen `assertThat` used in that way before, and it is probably not that common.